x/wasm/ioutils/ioutil_test.go

```diff
@@ -23,6 +23,51 @@ func TestUncompress(t *testing.T) {
 	wasmGzipped, err := os.ReadFile("../keeper/testdata/hackatom.wasm.gzip")
 	require.NoError(t, err)
 
+	specs := map[string]struct {
+		src       []byte
+		expError  error
+		expResult []byte
+	}{
+		"handle wasm compressed": {
+			src:       wasmGzipped,
+			expResult: wasmRaw,
+		},
+		"handle gzip identifier only": {
+			src:      gzipIdent,
+			expError: io.ErrUnexpectedEOF,
+		},
+		"handle broken gzip": {
+			src:      append(gzipIdent, byte(0x1)),
+			expError: io.ErrUnexpectedEOF,
+		},
+		"handle incomplete gzip": {
+			src:      wasmGzipped[:len(wasmGzipped)-5],
+			expError: io.ErrUnexpectedEOF,
+		},
+		"handle limit gzip output": {
+			src:       asGzip(bytes.Repeat([]byte{0x1}, 400_000)),
+			expResult: bytes.Repeat([]byte{0x1}, 400_000),
+		},
+	}
+	for msg, spec := range specs {
+		t.Run(msg, func(t *testing.T) {
+			r, err := Uncompress(spec.src)
+			require.True(t, errors.Is(spec.expError, err), "exp %v got %+v", spec.expError, err)
+			if spec.expError != nil {
+				return
+			}
+			assert.Equal(t, spec.expResult, r)
+		})
+	}
+}
```

Comment:
good test. But we would need one on the snapshotter to ensure a file > MaxWasmSize is accepted.